### PR TITLE
Support QuickSpec.current for SwiftPM builds as well on Darwin

### PR DESCRIPTION
**Pull request by ikesyo**
Ref: #645

Linux support will be added in a separate PR.

And also this is a preparation for #680 which uses `QuickSpec.current`.

 - [x] Does this have tests? Already exists
 - [x] Does this have documentation? Already exists
 - [x] Does this break the public API (Requires major version bump)? No
 - [x] Is this a new feature (Requires minor version bump)? Yes

### CODE REVIEW

File: Sources/Quick/QuickSpec.swift [hunk 1]
```diff
@@ -72,12 +84,13 @@ open class QuickSpec: QuickSpecBase {
...
     private static func addInstanceMethod(for example: Example, classSelectorNames selectorNames : inout Set<String>) -> Selector {
-        let block: @convention(block) (QuickSpec) -> Void = { _ in
+        let block: @convention(block) (QuickSpec) -> Void = { spec in
+            spec.example = example
             example.run()
         }
         let implementation = imp_implementationWithBlock(block as Any)
 
-        let originalName = example.name
+        let originalName = example.name.c99ExtendedIdentifier
```
Comment (ikesyo) [inline Sources/Quick/QuickSpec.swift]: There had been incompatibility with Xcode builds so fixed it.

https://github.com/Quick/Quick/blob/7e6b060b41e0110a5be1f4af324000acbfc21ae7/Sources/QuickObjectiveC/QuickSpec.m#L113

**Comment by ikesyo** [comment]
https://github.com/Quick/Quick/pull/634#issuecomment-480521569

I don't have an access to @QuickBot account, we might need to disable running danger (for a while).

**Comment by phatblat** [comment]
Opened #849 to track the danger/quickbot issue.

**Comment by ikesyo** [comment]
Thanks @phatblat for the review and filing the issue!

**Pull request merged**